lib/libutee/include/tee_api_defines_extensions.h

```diff
@@ -7,6 +7,10 @@
 #ifndef TEE_API_DEFINES_EXTENSIONS_H
 #define TEE_API_DEFINES_EXTENSIONS_H
 
+/* RSA signatures with MD5 hash */
+#define TEE_ALG_RSASSA_PKCS1_PSS_MGF1_MD5       0x70111930
```

Comment:
Implementation-defined algorithm identifiers should be in the range 0xF0000000 – 0xF0FFFFFF.

Reply:
I will make an update as soon as possible
Thank you

Reply:
Hello @jenswi-linaro . I am having some issues with this range of algorithm identifiers. Should I adapt all necessary optee-os code and add very specific algorithm code to make it work ? One small example of this is :  reworking all syscall_cryp_state_alloc() since TEE_ALG_GET_CLASS(algo) will be returning 0xF for that range and that is not defined to be an operation (I could define it but it will be a definition for all the 0xF0000000 – 0xF0FFFFFF identifier range ). Could you please tell what is the right way to do this implementation ?

Reply:
> One small example of this is : reworking all syscall_cryp_state_alloc() since TEE_ALG_GET_CLASS(algo) will be returning 0xF for that range and that is not defined to be an operation (I could define it but it will be a definition for all the 0xF0000000 – 0xF0FFFFFF identifier range )

`TEE_ALG_GET_CLASS()` is implemented as a static inline function: `__tee_alg_get_class()`. You can easily return whatever value is needed, and only for that particular algorithm identifier. Same for `TEE_ALG_GET_MAIN_ALG()` and possibly other things if needed.

See https://github.com/OP-TEE/optee_os/blob/3.20.0/lib/libutee/include/utee_defines.h#L57-L71.

Reply:
Thank you, it is clear to me now. I will make an update